lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2231,6 +2231,22 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
     // For llvm.fmuladd.* fusion is not guaranteed. If a fused multiply-add
     // is required the corresponding llvm.fma.* intrinsic function should be
     // used instead.
+    if (!checkTypeForSPIRVExtendedInstLowering(II, BM))
+      break;
+
+    // If allowed, let's replace llvm.fmuladd.* with mad from OpenCL extended
+    // instruction set, as it has the same semantic for FULL_PROFILE OpenCL
+    // devices (and for EMBEDDED_PROFILE precision is implementaiton-defined).
```

Comment:
```suggestion
    // devices (and for EMBEDDED_PROFILE precision is implementation-defined).
```

Reply:
Or even:
```suggestion
    // devices (implementation-defined for EMBEDDED_PROFILE precision ).
```

Reply:
Went with suggestion from Artem in 05d38e03deb11d274068805048ddbab164455924